recbole/data/dataloader/abstract_dataloader.py

```diff
@@ -137,7 +137,7 @@ def _set_neg_sample_args(self, config, dataset, dl_format, neg_sample_args):
         self.times = 1
         if (
             self.neg_sample_args["distribution"] == "uniform"
-            or "popularity"
+            or self.neg_sample_args["distribution"] == "popularity"
             and self.neg_sample_args["sample_num"] != "none"
```

Comment:
Hi @zhengbw0324,

This fix is still incorrect. When `distribution` is `uniform`, the `if` condition always returns `True` no matter what `sample_num` is.

So I suggest using the method mentioned in the PR:https://github.com/RUCAIBox/RecBole/pull/1863.
Or you can put the `self.neg_sample_args["distribution"] == "uniform" or self.neg_sample_args["distribution"] == "popularity"` into a parenthesis, i.e.,

```python
if (
      ( self.neg_sample_args["distribution"] == "uniform" 
        or self.neg_sample_args["distribution"] == "popularity" )     # <- like this
        and self.neg_sample_args["sample_num"] != "none"
):
    ...
```

Thanks!